sdk/python/lib/pulumi/automation/_workspace.py

```diff
@@ -52,23 +52,32 @@ def __init__(
         self.resource_count = resource_count
         self.url = url
 
-
+class TokenInformation:
+    """Information about the token that was used to authenticate the current user."""
+    name: str
+    organization: Optional[str]
+    team: Optional[str]
+    
 class WhoAmIResult:
     """The currently logged-in Pulumi identity."""
 
     user: str
     url: Optional[str]
     organizations: Optional[List[str]]
-
+    tokenInformation: Optional[TokenInformation]
+    
     def __init__(
         self,
         user: str,
         url: Optional[str] = None,
         organizations: Optional[List[str]] = None,
+        tokenInformation: Optional[TokenInformation] = None,
```

Comment:
This isn't python style naming (should be token_information) but I can see why the splat directly from JSON data requires this at the moment.
I'm going to raise a separate PR to stop doing that splat (because it causes issues if new unexpected data is in the JSON), once that's merged I think this can be rebased on that and renamed to token_information.

Reply:
You are absolutely right. the only reason its in camel case is to match the returned json data. 
Please comment your PR number here after opening it - so it'll be easier to track. 
Thanks!

Reply:
https://github.com/pulumi/pulumi/pull/17770

Suggest we get that merged, then you can rebase on that and we'll get this merged in after.

Reply:
@Frassle I rebased and added the token info in snake_case to the PR. please re-review. thanks!